libraries/shared/watcher/storage_watcher.go

```diff
@@ -89,34 +89,21 @@ func (watcher StorageWatcher) Execute() error {
 }
 
 func (watcher StorageWatcher) transformDiffs() error {
-	diffs := make(chan types.PersistedDiff)
-	errs := make(chan error)
-	done := make(chan bool)
-
-	defer close(diffs)
-	defer close(errs)
-	defer close(done)
-
-	go watcher.StorageDiffRepository.GetNewDiffs(diffs, errs, done)
```

Comment:
I'm trying to make sure I understand this change - was the only reason we were running `GetNewDiffs` in a go routine previously so that we could make sure not to block the transformation of diffs while we were getting new diffs from the cursor?

And now, since we're limiting what the query is returning and not using the cursor, we'll be doing the two operations synchronously because hopefully limiting the query to 500 will be much quicker than getting the whole set of diffs. Is that right?

Reply:
Thinking about this a bit more... I think it probably make sense to try out this synchronous approach first. But I wonder if in the future we move back to running `GetNewDiffs` in a go routine with several workers, and put those diffs on a diff channel. I wonder if this may give up a performance bump as well. Though, probably not drastic. Because that way, `transformDiffs` will never have to wait for `GetStorageDiffs` to return diffs - it will always have them to pop off the channel.

Reply:
Yeah - you got it! Also hoping to avoid keeping a connection alive that's running through millions of rows

Reply:
Something weird going on for me and missed your more recent comment, but that makes sense to me. I'd love to be able to be fetching diffs and transforming diffs simultaneously, especially if we can be doing those things concurrently. Only concern would be making sure we're not keeping a long running query alive so long that it interferes with autovacuum